### PR TITLE
Maintainance

### PR DESCRIPTION
**Pull request by kaqmak**
Upgrading all packages and bumped compatible python version to include 3.11.
precommit now uses ruff and thus isort and flake8 is no longer needed.
RegFeatureSelector was entirely removed, since the tests failed: I had the to change the boston_dataset to california housing prices, but for some reason feature_selection tests were now failing. Since it was already deemed deprecated I have now entirely removed it.

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -1,42 +1,42 @@
 [tool.poetry]
 name = "dfds_ds_toolbox"
-version = "0.10.2"
+version = "0.10.3"
```
Comment (pacanada) [inline pyproject.toml:3]: I guess there are some breaking changes since we are removing `feat_selector.py`. maybe `0.11.0`

**Comment by pacanada** [comment]
I think there is a reference to the remove feature selection module in `docs` : https://github.com/dfds-data/dfds-ds-toolbox/blob/e83165ad2ec29813e3e28cd24ee5405648f540c1/docs/usage.rst#:~:text=from%20sklearn.pipeline,VarianceThreshold())%2C%0A%20%20%20%20%20%20%20%20(%27model%27%2C%20RegSwitcher())%0A%20%20%20%20%5D%0A)

**Comment by pacanada** [comment]
Test are failing on the installation of dependencies, seems to be poetry compatibility (the pr workflow is set to poetry 1.1 ). I would also include python 3.11 for the tests

File: pyproject.toml (hunk 1 above)
Comment (kaqmak) [reply pyproject.toml:3]: Done

**Review by pacanada — CHANGES_REQUESTED** [review]
LGTM, only the comments i made about the leftovers in docs and the python tag

File: pyproject.toml [hunk 2]
```diff
@@ -1,42 +1,42 @@
...
-        "Programming Language :: Python :: 3.10",
-        "License :: OSI Approved :: MIT License",
-]
-packages = [
-    { include = "dfds_ds_toolbox" }
+  "Development Status :: 3 - Alpha",
+  "Programming Language :: Python :: 3 :: Only",
+  "Programming Language :: Python :: 3.8",
+  "Programming Language :: Python :: 3.9",
+  "Programming Language :: Python :: 3.10",
```
[the remark this answers is not in the document]
Comment (kaqmak) [reply pyproject.toml:12]: yes. Changed to include 3.11 and 3.10

**Comment by kaqmak** [comment]
> There is a reference to the remove feature selection module in `docs` : https://github.com/dfds-data/dfds-ds-toolbox/blob/e83165ad2ec29813e3e28cd24ee5405648f540c1/docs/usage.rst#:~:text=from%20sklearn.pipeline,VarianceThreshold())%2C%0A%20%20%20%20%20%20%20%20(%27model%27%2C%20RegSwitcher())%0A%20%20%20%20%5D%0A)



> There is a reference to the remove feature selection module in `docs` : https://github.com/dfds-data/dfds-ds-toolbox/blob/e83165ad2ec29813e3e28cd24ee5405648f540c1/docs/usage.rst#:~:text=from%20sklearn.pipeline,VarianceThreshold())%2C%0A%20%20%20%20%20%20%20%20(%27model%27%2C%20RegSwitcher())%0A%20%20%20%20%5D%0A)

I don't see where that is. Can you give me a line number?

**Review by pacanada — APPROVED** [review]

**Pull request merged**